### ISSUE TITLE
Model training in the GPflow-2-version of the coregionalisation notebook is very slow

### ISSUE THREAD

**Issue by fleibfried** [GPflow/GPflow#1090]
System information:

* Python version: 3.7.4
* TensorFlow installed from binary
* TensorFlow version: 2.0.0-dev20191002
* GPflow installed from source
* GPflow version: 1.9.999

... _Describe the current behavior_
Training a model with a coregionalisation kernel takes much longer compared to the GPflow-1-version.

... _Describe the expected behavior_
Faster model training.

... _Code to reproduce the issue Provide a reproducible test case that is the bare minimum necessary to generate the problem._
see branch 'felix/coregionalisation' in the directory 'notebooks2/advanced'

**Comment by st--** [comment]
(See #1072 for the gpflow2 version of the coregionalisation.ipynb notebook)

**Comment by st--** [comment]
It looks like it may be some weird interaction with the (L-)BFGS(-B) optimizer default -- it seems to work much better (faster, in terms of wallclock speed if not necessarily in loss vs iterations) with conjugate gradients (`method='CG'` argument to Scipy().minimize())...

**Comment by st--** [comment]
Turns out the issue was that the default method of Scipy optimizer had changed from explicitly selecting `L-BFGS-B` in gpflow1 to letting scipy.minimize pick its default which ended up being `BFGS` (see #1111). BFGS constructs a full Hessian, which explains why this notebook was so slow (and why it used up so much memory when running). I've now fixed the method to be L-BFGS-B, so regardless of merging #1111 this notebook is now fast & gives the answers that we expect.
